op-node/rollup/derive/batch_queue.go

```diff
@@ -122,196 +97,140 @@ func (bq *BatchQueue) AddBatch(batch *BatchData) {
 	if len(bq.l1Blocks) == 0 {
 		panic(fmt.Errorf("cannot add batch with timestamp %d, no origin was prepared", batch.Timestamp))
 	}
-	bq.log.Trace("queuing batch", "origin", bq.progress.Origin, "tx_count", len(batch.Transactions), "timestamp", batch.Timestamp)
+	log := bq.log.New(
+		"current_l1", bq.progress.Origin,
+		"batch_timestamp", batch.Timestamp,
+		"batch_epoch", batch.Epoch(),
+		"txs", len(batch.Transactions),
+	)
+	log.Trace("queuing batch")
 
 	data := BatchWithL1InclusionBlock{
 		L1InclusionBlock: bq.progress.Origin,
 		Batch:            batch,
 	}
-	batches, ok := bq.batchesByTimestamp[batch.Timestamp]
-	// Filter complete duplicates. This step is not strictly needed as we always append, but it is nice to avoid lots of spam.
-	if ok {
-		for _, b := range batches {
-			if b.Batch.Timestamp == batch.Timestamp && b.Batch.Epoch() == batch.Epoch() {
-				bq.log.Warn("duplicate batch", "epoch", batch.Epoch(), "timestamp", batch.Timestamp, "txs", len(batch.Transactions))
-				return
-			}
-		}
-	} else {
-		bq.log.Debug("First seen batch", "epoch", batch.Epoch(), "timestamp", batch.Timestamp, "txs", len(batch.Transactions))
-	}
-	// May have duplicate block numbers or individual fields, but have limited complete duplicates
-	bq.batchesByTimestamp[batch.Timestamp] = append(batches, &data)
-}
-
-// validExtension determines if a batch follows the previous attributes
-func (bq *BatchQueue) validExtension(batch *BatchWithL1InclusionBlock, prevTime, prevEpoch uint64) (valid bool, err error) {
-	if batch.Batch.Timestamp != prevTime+bq.config.BlockTime {
-		bq.log.Debug("Batch does not extend the block time properly", "time", batch.Batch.Timestamp, "prev_time", prevTime)
-		return false, nil
-	}
-	if batch.Batch.EpochNum != rollup.Epoch(prevEpoch) && batch.Batch.EpochNum != rollup.Epoch(prevEpoch+1) {
-		bq.log.Debug("Batch does not extend the epoch properly", "epoch", batch.Batch.EpochNum, "prev_epoch", prevEpoch)
-		return false, nil
-	}
-	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-	l1BlockRef, err := bq.dl.L1BlockRefByNumber(ctx, batch.Batch.Epoch().Number)
-	cancel()
-	if err != nil {
-		return false, err
-	}
-
-	if l1BlockRef.Hash != batch.Batch.EpochHash {
-		bq.log.Debug("Batch epoch hash does not match expected L1 block hash", "batch_epoch", batch.Batch.Epoch(), "expected", l1BlockRef.ID())
-		return false, nil
+	validity := CheckBatch(bq.config, bq.log, bq.l1Blocks, bq.next.SafeL2Head(), &data)
```

Comment:
Passing a logger into `CheckBatch` feels kinda weird to me but I get it, why use `bq.log` here instead of the defined `log` above?

Reply:
`CheckBatch` already adds the same logging details, so that the details are there for every caller.

Reply:
I'll try clean up the logging a bit

Reply:
Removed the logging there, in favor of making the `CheckBatch` logging, which covers all cases now, and has warn-level if the batch needs to be dropped.